src/Pickles/Pickles.UserInterface/ViewModel/MainViewModel.cs

```diff
@@ -485,6 +505,44 @@ private bool CanGenerate()
                    && this.isLanguageValid;
         }
 
+        private void DoGenerateCommandLine(CommandGeneratorBase generator)
+        {
+            var model = this.CreateMainModel();
+            var language = this.selectedLanguage.TwoLetterISOLanguageName;
+            var commands = generator.Generate(model, language);
+            Clipboard.SetText(commands);
+            MessageBox.Show("Copied the following commands into the clipboard:\n\n" + commands, "Info", MessageBoxButton.OK, MessageBoxImage.Information);
```

Comment:
This I don't like at all - it breaks the decoupling between the viewmodel and the view.
I would like to remove it, but then the user will have no feedback that anything happened when they click on one of the command-generating buttons. So what I suggest is that you include a logger in the class (`private static readonly Logger Log = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType.Name);`) and write the text into the logger. Then the user will see the result of their action in the logging grid.

Reply:
Agreed and fixed